### ISSUE TITLE
AdamW Optimizer support

### ISSUE THREAD

**Issue by salwaghanim** [THU-MIG/yolov10#103]
Hello,
do you currently support AdamW optimizer? if Not do you a time estimation for when AdamW or other optimizers will be supported
I have trained a custom yolov10 nano and small models. comparing to yolov8 models on the same dataset Their is a significant speed difference. However Yolov8 achieved better precision. In my experience AdamW is better suited when dealing with custom dataset of low number of classes. 
for reference here is my training code for achieving better precision with yolov8 the only variable that might boosted yolov8 performance is AdamW ps. I increased patience to preserve the same learning rate (achieve convergence with the same number of epochs) and apply close_mosaic

```
from ultralytics import YOLO,YOLOv10
import argparse
import os


def main():
    parser = argparse.ArgumentParser(
        description="Yolo Trainer"
    )
    parser.add_argument("-n", "--epochs", type=int, default=500, help="epochs value.")
    parser.add_argument("--name", help="name of the model")
    parser.add_argument(
        "--resume",
        action="store_true",
        help="Whether to resume training from a .pt checkpoint",
    )
    parser.add_argument(
        "--yolov10",
        action="store_true",
        help=" use to train you are training a yolov10 model",
    )

    args = parser.parse_args()

    YOLOV10_MODELS_PATH = r"/media/experement/Data/A/Current_setup_data/yolov10/"
    model = YOLO("yolov8s.pt")

    if args.yolov10:
        model_path = os.path.join(YOLOV10_MODELS_PATH,'yolov10n.pt')
        model = YOLOv10(model_path)


    if args.resume:
        results = model.train(resume=True)

    else:
        results = model.train(
            data="/media/experement/Data/A/Datasets/To_Assess/Kites_v2/data.yaml",
            optimizer="AdamW",
            epochs=args.epochs,
            batch=-1,
            imgsz=640,
            device=0,
            name=args.name,
            amp=True,
            cache=True,
            plots=True,
            close_mosaic=10,
            patience=150,
        )


if __name__ == "__main__":
    main()

```

**Comment by jameslahm** [comment]
Thanks for your interest! The Adam optimizer is supported in this codebase. You could specify the `optimizer='Adam'` like YOLOv8. Besides, we would also like to confirm if you are using our updated [checkpoint](https://github.com/THU-MIG/yolov10/releases/tag/v1.1)? The updated checkpoints are aligned with those of YOLOv8 and include additional training hyperparameters. Could this possibly affect the results? Thanks.

**Comment by salwaghanim** [comment]
Thanks I Will checkout the updated checkpoints, and I have managed to train with adam Optimizer using the python SDK, since the bash is not recognizing the --optimizer argument.
I will close this issue Once I confirm that the precision  is  higher than yolov8 so Everyone benefits from it.
ps. I have read that yolov8 is better for Far objects from the camera. do you have any recommendation for such cases (small objects or far from camera objects.)

**Comment by jameslahm** [comment]
Thanks. We provide some clarification in #100 and https://huggingface.co/spaces/SkalskiP/YOLO-ARENA/discussions/1. 
> Moreover, we want to clarify that due to that YOLOv10 adopts the different training strategy with YOLOv8 and YOLOv9, it may have different favorable confidence threshold to detect objects. Therefore, comparing the results of YOLOv10 with those of YOLOv8 and YOLOv9 on detecting small objects may be less appropriate with the same threshold. Besides, we note that different thresholds will not have the adverse impact on the inference latency of YOLOv10 because it does not rely on NMS. Therefore, we can freely set a smaller threshold for YOLOv10 to detect more small objects. For example,
>  ```
>  wget https://skalskip-yolo-arena.hf.space/file=/tmp/gradio/048ecf37df927d6d7ed5528076f4133147f797d2/people-walking.png
>  yolo export model=yolov10m.pt format=onnx opset=13 simplify half=True device=0
>  yolo predict model=yolov10m.onnx source=people-walking.png half conf=0.05
>  ```
> The inference result is below
![](https://cdn-uploads.huggingface.co/production/uploads/628ece6054698ce61d1e7be3/rPBmtofj4zx9TPW5_lAqP.png)

**Comment by salwaghanim** [comment]
Thanks for your fast reply! I have initiated the yolov10 training on the latest checkpoint. I will report back the precision difference when the training is complete. Have a great day!

**Comment by salwaghanim** [comment]
The performance is now better however yolov8 is still  better in my use case. Moreover, and this is really weird yolov8 inference is faster.  I Can think of possible reason, I had to downgrade to opencv 4.6 are you using the winograd matrix multiplication? I know that this function usually boost performance while preserving the accuracy and its supported in opencv 4.9

**Comment by jameslahm** [comment]
Thanks. Regarding the performance, could this comment https://github.com/THU-MIG/yolov10/issues/111#issuecomment-2137277092 help? Regarding the inference speed, could you please more details about the measurement? Thank you!

**Comment by salwaghanim** [comment]
Sure here is my yolov10  inference script:

```
import cv2
from ultralytics import YOLOv10
import supervision as sv

# Load YOLOv10 model
model = YOLOv10('/media/experement/Data/A/Current_setup_data/yolov8/runs/detect/kites_yolov10_n/weights/best.pt')  # nano

# Load video
video_path = "/media/experement/Data/A/Datasets/deployment ready/testing_videos/kites_1.mp4"
cap = cv2.VideoCapture(video_path)

# Check if the video was opened successfully
if not cap.isOpened():
    print(f"Error: Could not open video {video_path}")
    exit()

# Initialize annotators
bounding_box_annotator = sv.BoundingBoxAnnotator()
label_annotator = sv.LabelAnnotator()

# Initialize variables for FPS calculation
fps = 0
frame_count = 0
start_time = cv2.getTickCount()

# Read video frame by frame
while cap.isOpened():
    ret, frame = cap.read()
    if not ret:
        break

    # Perform inference on the frame
    results = model(frame, conf=0.25)[0]
    detections = sv.Detections.from_ultralytics(results)

    # Annotate frame with bounding boxes and labels
    annotated_frame = bounding_box_annotator.annotate(scene=frame, detections=detections)
    annotated_frame = label_annotator.annotate(scene=annotated_frame, detections=detections)

    # Calculate FPS
    frame_count += 1
    end_time = cv2.getTickCount()
    time_spent = (end_time - start_time) / cv2.getTickFrequency()
    if time_spent > 1:  # Update FPS every second
        fps = frame_count / time_spent
        frame_count = 0
        start_time = cv2.getTickCount()

    # Display FPS on the frame
    cv2.putText(annotated_frame, f"FPS: {fps:.2f}", (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 0), 2)

    # Display annotated frame
    cv2.imshow("YOLOv10 Inference", annotated_frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()
```
yolov8 inference:

```
import cv2
from ultralytics import YOLO

# Load the YOLOv8 model
model = YOLO('/media/experement/Data/A/Current_setup_data/yolov8/runs/detect/Gathered_kitesv2_n/weights/best.pt')

# Open the video file
video_path = r'/media/experement/Data/A/Datasets/deployment ready/testing_videos/kites_1.mp4'
cap = cv2.VideoCapture(video_path)

# Check if the video was opened successfully
if not cap.isOpened():
    print(f"Error: Could not open video {video_path}")
    exit()

# Initialize variables for FPS calculation
fps = 0
frame_count = 0
start_time = cv2.getTickCount()

# Loop through the video frames
while cap.isOpened():
    # Read a frame from the video
    success, frame = cap.read()

    if success:
        # Run YOLOv8 inference on the frame
        results = model(frame)

        # Visualize the results on the frame
        annotated_frame = results[0].plot()

        # Calculate FPS
        frame_count += 1
        end_time = cv2.getTickCount()
        time_spent = (end_time - start_time) / cv2.getTickFrequency()
        if time_spent > 1:  # Update FPS every second
            fps = frame_count / time_spent
            frame_count = 0
            start_time = cv2.getTickCount()

        # Display FPS on the frame
        cv2.putText(annotated_frame, f"FPS: {fps:.2f}", (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 0), 2)

        # Display the annotated frame
        cv2.imshow("YOLOv8 Inference", annotated_frame)

        # Press 'P' to pause the video until any key is pressed
        key = cv2.waitKey(1)
        if key & 0xFF == ord('p'):
            cv2.waitKey(-1)  # Wait until any key is pressed
        elif key & 0xFF == ord('q'):
            break
    else:
        # Break the loop if the end of the video is reached
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()

```
yolov8n is faster by 5fps in average.

**Comment by jameslahm** [comment]
Thanks! Could you please try to use the exported format, e.g., TensorRT engine, to benchmark the speed?

**Comment by jameslahm** [comment]
Please feel free to reopen this issue if you have further questions.
